src/Tempest/Mailer/src/EmailBuilder.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Mailer;
+
+use Tempest\Mailer\Components\Address\Address;
+use Tempest\Mailer\Components\Address\AddressCollection;
+
+final class EmailBuilder
```

Comment:
IDK… we talked about this for requests as well. I understand that you like the pattern, but I don't think it's all that useful for most cases. 

- We could make `Email` itself immutable, add these methods on it, and always return a clone of the object. 
- Or we could not have these methods at all, why would you want to change something after the first time constructing an email/builder anyway?

My preference is to add immutable builder methods on `Email` directly.

Reply:
> We could make Email itself immutable, add these methods on it, and always return a clone of the object.

I'm going to shock you real quick... the way PSR handles immutable objects was informed by limitations in PHP at the time and there was much debate in the drafting of it. IMHO, it's just wrong because it's not truly immutable.

If a transport gets passed an email, it could modify the contents of the email. That shouldn't be possible.